### PR TITLE
Provide instructions for using Docker on Travis CI

### PR DESCRIPTION
**Pull request by jayvdb**

### CODE REVIEW

**Comment by jayvdb** [comment]
This is part of https://github.com/coala-analyzer/docker-coala-base/issues/15

**Comment by sils1297** [comment]
Hi, was this approach tested? Also CC @underyx for docker knowledge.

File: Users/Docker_Image.rst [hunk 1]
```diff
@@ -45,3 +45,23 @@ for the ``gitlab-ci-multi-runner`` runner:
...
+on your code with a ``.travis.yml``, like this:
+
+::
+
+    language: generic
+    sudo: required
+    services:
+      - docker
+    script:
+      - docker run -v $(pwd):/app -w /app coala/base coala-ci
```
Comment (sils1297) [inline Users/Docker_Image.rst]: we may want to note that this same this very command can be used for any CI where docker is present so this isn't limited to travis

**Comment by jayvdb** [comment]
I agree that the `docker` command could be separately documented at the top as general guidance for any CI where docker is present (or installed manually)

Yes, I tested it on Travis.

I've found that it can be shorted to three lines with very little surrounding syntax:
```yaml
language: generic
services: docker
script: docker run --volume=$PWD:$PWD --workdir=$PWD coala/base coala-ci
```

I am using $PWD in the above, for both sides of the mount point, as I think that makes it easier to understand what is happening.

However it could cause a bit of confusion about whether a part of the log is running on the travis virtual environment, or the Docker virtual environment.  Maybe this would be better:

```bash
$ docker run --volume=$PWD:/mnt/$PWD --workdir=/mnt/$PWD coala/base coala-ci
```

**Comment by sils1297** [comment]
Can we have this shortened version in the docs?

**Comment by jayvdb** [comment]
Yes, I was planning to use the shorter one.

@sils1297, would appreciate your opinion on using `$PWD` or `/mnt/$PWD` as the mount point inside of the Docker image, instead of `/app` as you suggested at https://github.com/coala-analyzer/docker-coala-base/issues/15#issuecomment-245864468 .

If you would prefer `/app`, I'm happy to stick with that.

**Comment by underyx** [comment]
@jayvdb I'd much prefer `/app`. I don't see why cloning the directory structure would make it easier to understand what `--volume` does, and `/app` is sort of a standard path for user code inside docker containers.

**Comment by sils1297** [comment]
@underyx can you look over the docker stuff? It looks good to me.

**Comment by underyx** [comment]
Yep, looks good, ack https://github.com/coala-analyzer/documentation/pull/100/commits/3ca120d32a79739785e00f3780d1692ad3b56826

Thank you very much for the contribution @jayvdb!

**Comment by underyx** [comment]
@rultor merge

**Pull request merged**